streams/src/main/java/org/apache/kafka/streams/processor/assignment/assignors/StickyTaskAssignor.java

```diff
@@ -278,7 +278,7 @@ public void processOptimizedAssignments(final Map<ProcessId, KafkaStreamsAssignm
 
             for (final Map.Entry<ProcessId, KafkaStreamsAssignment> entry : optimizedAssignments.entrySet()) {
                 final ProcessId processId = entry.getKey();
-                final Set<AssignedTask> assignedTasks = optimizedAssignments.get(processId).assignment();
+                final Set<AssignedTask> assignedTasks = new HashSet<>(optimizedAssignments.get(processId).tasks().values());
```

Comment:
This can be a followup PR, but one of the nice things about making KafkaStreamsAssignment mutable is that we should be able to get rid of the `newAssignments` field altogether and stop converting back and forth between the KafkaStreamsAssignment and raw Sets so that we can add/remove tasks. 

In addition to the obvious code simplification that should result from this change, it should save us a lot of copying things into various different data structures and reduce a lot of overhead

Reply:
Created a ticket for it.